src/cast_core/src/RunCommand.cpp

```diff
@@ -208,13 +208,110 @@ class DatabaseOperation {
     bool _awaitStepdown;
 };
 
+// Class to parse and hold the OnlyRunInInstance configuration. This option may be used to only
+// run under certain configurations Available configurations are specified in Options.
+class InstanceTypeFilter {
+public:
+    struct Options {
+        static constexpr std::string_view kStandalone = "standalone";
+        static constexpr std::string_view kSharded = "sharded";
+        static constexpr std::string_view kReplicaSet = "replica_set";
+
+        static std::string getValidOptionsListAsString() {
+            std::stringstream listString;
+            listString << kStandalone << ", " << kSharded << ", " << kReplicaSet;
+            return listString.str();
+        }
+
+        static bool isValid(std::string_view instanceType) {
+            return instanceType == kStandalone || instanceType == kSharded
+                    || instanceType == kReplicaSet;
+        }
+
+        static bool isValidList(const std::vector<std::string>& options) {
+            for(auto &option : options) {
+                if(!isValid(option)) {
+                    return false;
+                }
+            }
+            return true;
+        }
+    };
+
+    InstanceTypeFilter(PhaseContext &context, mongocxx::pool::entry& client) {
+        try {
+            _onlyRunInInstancesContext = context.getPlural<std::string>(
+                "OnlyRunInInstance", "OnlyRunInInstances", [&](const Node& node) {
+                    return node.to<std::string>();
+            });
+        } catch(...) {
+            // Exception might be due to keys not found or other errors. If its due to keys not
+            // found ignore the expception. Otherwise rethrow.
+            if(context["OnlyRunInInstance"] || context["OnlyRunInInstance"]) {
+                std::rethrow_exception(std::current_exception());
+            }
+        }
+
+        if(_onlyRunInInstancesContext.size()) {
+            if(!Options::isValidList(_onlyRunInInstancesContext)) {
+                std::stringstream errorMsg;
+                errorMsg << "OnlyRunInInstance or OnlyRunInInstances valid values are: "
+                    << Options::getValidOptionsListAsString();
+                throw InvalidConfigurationException(errorMsg.str());
+            }
+
+            _adminDB = (*client)["admin"];
+        }
+    }
+
+    // Indicates if operations should be run. If there is no OnlyRunInInstance configuration or one
+    // of the specified options matches the client instance type, returns true. Otherwise returns
+    // false.
+    bool shouldRun() {
+        if(_type.length() == 0 && _onlyRunInInstancesContext.size()) {
+            _type = getInstanceType(_adminDB);
+            _typeFoundInConfig = false;
+            for(const auto &configInstanceType : _onlyRunInInstancesContext) {
```

Comment:
Should we go ahead and quit out if there is no configured instance type, before we make the hello query to determine it (possibly pointlessly?)

Reply:
This is already what is done. See line [271](https://github.com/mongodb/genny/pull/651/files#diff-05612e944a021e56717199e577fcac940d40eb7b838cf41da8361ed32fd9ed35R271). getInstanceType is only called when _onlyRunInInstancesContext.size() is not 0.

Reply:
I changed it any ways to do an early return, so the code is easier to understand.

Reply:
Thank you!